File: python/algorithms/easy/appleAndOrange.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-07-13
Last modified: 2026-07-13
Description: Apple and Orange - HackerRank
Test description: https://www.hackerrank.com/challenges/apple-and-orange/problem
"""
#---------- functions ----------
def countApplesAndOranges(s: int, t: int, a: int, b: int, apples: list[int], oranges: list[int]) -> None:
    """
    Counts the number of apples and oranges that fall on Sam's house.

    Parameters:
        s (int): Start position of Sam's house.
        t (int): End position of Sam's house.
        a (int): Position of the apple tree.
        b (int): Position of the orange tree.
        apples (list[int]): List of distances that apples fall from the tree.
        oranges (list[int]): List of distances that oranges fall from the tree.
    """
    countApples = 0
    countOranges = 0

    for apple in apples:
        reach = a + apple
        if s <= reach <= t:
            countApples += 1

    for orange in oranges:
        reach = b + orange
        if s <= reach <= t:
            countOranges += 1

    print(countApples)
    print(countOranges)

File: typescript-javascript/algorithms/easy/appleAndOrange.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-07-13
 * Last modified: 2026-07-13
 * @description Apple and Orange - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/apple-and-orange/problem}
 */
//---------- functions ----------
/**
 * Counts the number of apples and oranges that fall on Sam's house.
 * 
 * @param {number} s
 * @param {number} t
 * @param {number} a
 * @param {number} b
 * @param {number[]} apples
 * @param {number[]} oranges
 */
function countApplesAndOranges(s: number, t: number, a: number, b: number, apples: number[], oranges: number[]): void 
{
    let countApples = 0, countOranges = 0;
    
    for(const apple of apples)
    { 
        let reach = a + apple;
        if(s <= reach && reach <= t)
        { countApples++; }   
    }//end for
    
    for(const orange of oranges)
    { 
        let reach = b + orange;
        if(s <= reach && reach <= t)
        { countOranges++; }
    }//end for
    
    console.log(countApples);
    console.log(countOranges);
}//end countApplesAndOranges()